### PR TITLE
Delete CocoaLumberjack

### PR DESCRIPTION
**Pull request by crazytonyli**
Similar to https://github.com/wordpress-mobile/WordPress-iOS-Shared/pull/325. I actually copied the main changes (the new logging functions) from that PR.

Most of the changes are replacing `DDLogXxx` with `WPAuthenticatorLogXxx`, which is added in [`WPAuthenticatorLogging.h`](https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/708/files#diff-a563a9925e21e4cabe222ca88e19fb552af352d3e05c3c4b78cc6da57976a18a).

**TODO**:
- [x] Update WordPressShared dependency after [2.0.0 is released](https://github.com/wordpress-mobile/WordPress-iOS-Shared/pull/326).
- [x] Update WordPressKit dependency after a new version is released.

---

- [x] I have considered if this change warrants release notes and have added them to the appropriate section in the `CHANGELOG.md` if necessary.

### CODE REVIEW

File: WordPressAuthenticator.podspec [hunk 1]
```diff
@@ -42,6 +41,6 @@ Pod::Spec.new do |s|
   # Use a loose restriction that allows both production and beta versions, up to the next major version.
   # If you want to update which of these is used, specify it in the host app.
   s.dependency 'WordPressUI', '~> 1.7-beta'
-  s.dependency 'WordPressKit', '~> 4.18-beta' # Don't change this until we hit 5.0 in WPKit
-  s.dependency 'WordPressShared', '~> 1.12-beta' # Don't change this until we hit 2.0 in WPShared
+  s.dependency 'WordPressKit', '~> 5.0.0-beta.1'
+  s.dependency 'WordPressShared', '~> 2.0.0-beta.1'
```
Comment (mokagio) [inline WordPressAuthenticator.podspec]: For a while, I though we had to either use `~> 2.0.0-beta.1` and be constrained to `2.0.x` versions, or `'~> 2.0', '>= 2.0.0-beta.1'` and pay the price of a more verbose specification to get all the `2.x` version.

I just learned that we can use `~> 2.0-beta` to get the best of both world. Incidentally, that's also the style used in the previous version of this code.

File: WordPressAuthenticatorTests/Logging/LoggingTests.swift [hunk 2]
```diff
@@ -0,0 +1,70 @@
+import XCTest
+
+@testable import WordPressAuthenticator
+
+private class CaptureLogs: NSObject, WordPressLoggingDelegate {
+    var verboseLogs = [String]()
+    var debugLogs = [String]()
+    var infoLogs = [String]()
+    var warningLogs = [String]()
+    var errorLogs = [String]()
```
Comment (mokagio) [inline WordPressAuthenticatorTests/Logging/LoggingTests.swift:10]: Same nitpick as [in WordPressKit-iOS](https://github.com/wordpress-mobile/WordPressKit-iOS/pull/550#discussion_r1033261108):

```suggestion
    private(set) var verboseLogs = [String]()
    private(set) var debugLogs = [String]()
    private(set) var infoLogs = [String]()
    private(set) var warningLogs = [String]()
    private(set) var errorLogs = [String]()
```

**Review by mokagio — COMMENTED** [review]
Changes looking good 👍

File: WordPressAuthenticatorTests/Logging/LoggingTests.swift (hunk 2 above)
Comment (crazytonyli) [reply WordPressAuthenticatorTests/Logging/LoggingTests.swift:10]: Thanks! I've taken your code suggestion.

**Review by mokagio — APPROVED** [review]

**Pull request merged**